### PR TITLE
avoid duplicates by not selecting all rows for global admins

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/samson

### CODE REVIEW

File: app/models/concerns/searchable.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module ClassMethods
     def search_by_criteria(criteria)
       scope = self
       scope = scope.search(criteria[:search]) if criteria[:search]
-      scope.order("#{sort_column(criteria[:sort])} #{sort_direction(criteria[:direction])}").page(criteria[:page])
+      scope.order("#{table_name}.#{sort_column(criteria[:sort])} #{sort_direction(criteria[:direction])}").page(criteria[:page])
```
Comment (grosser) [inline app/models/concerns/searchable.rb]: not needed, but now this can be used with group statements too ...

File: app/models/user.rb [hunk 2]
```diff
@@ -27,10 +27,11 @@ class User < ActiveRecord::Base
...
   scope :with_role, -> (role_id, project_id) {
-    joins('LEFT OUTER JOIN user_project_roles ON users.id = user_project_roles.user_id').
+    joins('LEFT JOIN user_project_roles ON users.id = user_project_roles.user_id').
       where(
-        '(user_project_roles.project_id = ? AND user_project_roles.role_id >= ?) OR users.role_id >= ?',
-        project_id, role_id, role_id
+        '(user_project_roles.project_id IS NULL AND users.role_id >= ?) OR ' \
+        '(user_project_roles.project_id = ? AND user_project_roles.role_id >= ?)',
+        role_id, project_id, role_id
       )
```
Comment (grosser) [inline app/models/user.rb]: before we selected users multiple times if they had multiple user_project_roles

**Comment by rikeoka** [comment]
Security filed JIRA: SAMSON-240 for this bug as well.

**Comment by grosser** [comment]
looks good / +1 ?

**Comment by rikeoka** [comment]
I noticed the same thing that I was seeing when I tried this approach before.

List of users ordered before filtering:
screen shot 2016-05-16 at 1 51 53 pm

With Search filter:
screen shot 2016-05-16 at 1 52 06 pm

Looking into if my test data was corrupted.

File: app/models/user.rb [hunk 3]
```diff
@@ -27,10 +27,11 @@ class User < ActiveRecord::Base
     where("name LIKE ? OR email LIKE ?", "%#{query}%", "%#{query}%")
   }
   scope :with_role, -> (role_id, project_id) {
-    joins('LEFT OUTER JOIN user_project_roles ON users.id = user_project_roles.user_id').
+    joins('LEFT JOIN user_project_roles ON users.id = user_project_roles.user_id').
```
Comment (rikeoka) [inline app/models/user.rb]: joins("LEFT OUTER JOIN user_project_roles ON users.id = user_project_roles.user_id AND user_project_roles.project_id = #{project_id.to_i}").
      where('users.role_id >= ? OR user_project_roles.role_id >= ?', role_id, role_id)

**Comment by grosser** [comment]
added test that fails without the changed sql ... good to go ?

**Comment by rikeoka** [comment]
Addresses that disappearing users bug I was noticing.  Not sure if I can give it the ok not being a core team member.

**Comment by grosser** [comment]
You probably know more about the user permissions then anyone else atm :D

**Comment by rikeoka** [comment]
👍

**Pull request merged**